src/features/attributeSelect.tsx

```diff
@@ -567,16 +682,119 @@ const TabsPanel = ({
+                            {Object.entries(TEMPLATE_MAP).map(([name, { description }]) => (
+                              <li key={name}>
+                                <span className="font-mono">{name}:</span>{" "}
+                                {description}
+                              </li>
+                            ))}
+                          </ul>
+                        </div>
+                      }
+                      placement="top"
+                    >           
+                      <Icon icon="info-sign" className="opacity-80" />   
+                    </Tooltip>
+                  </div>
+                  
+                  {selectedTemplate === "Custom Format" && (
+                    <div className="space-y-2">
+                      <FormGroup label="Pattern (regex)" className="m-0">
+                        <input
+                          className="bp3-input font-mono text-sm w-full"
+                          placeholder="E.g., \[\[(.*?)\]\]"
+                          value={customPattern}
+                          onChange={(e) => {
+                            const newValue = e.target.value;
+                            setCustomPattern(newValue);
+                            setInputSetting({
+                              blockUid: attributeUid,
+                              key: "customPattern",
+                              value: newValue,
+                            });
+                          }}
+                        />
+                      </FormGroup>
+                      
+                      <FormGroup label="Replacement" className="m-0">
+                        <input
+                          className="bp3-input font-mono text-sm w-full"
+                          placeholder="E.g., $1"
+                          value={customReplacement}
+                          onChange={(e) => {
+                            const newValue = e.target.value;
+                            setCustomReplacement(newValue);
+                            setInputSetting({
+                              blockUid: attributeUid,
+                              key: "customReplacement",
+                              value: newValue,
+                            });
+                          }}
+                        />
+                      </FormGroup>
+                      
+                      <div className="bg-gray-100 p-2 rounded text-sm">
+                        <div className="font-bold mb-1">Preview:</div>
+                        <div>
+                          <span className="font-bold">Input:</span> <span className="font-mono">[[Example]]</span>
+                        </div>
+                        <div>
+                          <span className="font-bold">Output:</span> <span className="font-mono">
+                            {customPattern ? 
+                              applyFormatting("[[Example]]", "Custom Format", customPattern, customReplacement) :
```

Comment:
If more than 2-3 params, use named params. They are more flexible, scalable, and self documenting.